src/main/java/picocli/CommandLine.java

```diff
@@ -6932,7 +6932,39 @@ boolean resemblesOption(String arg, Tracer tracer) {
                 if (tracer != null && tracer.isDebug()) {tracer.debug("'%s' %s an option: %d matching prefix chars out of %d option names%n", arg, (result ? "resembles" : "doesn't resemble"), count, optionsMap().size());}
                 return result;
             }
+
+            public ArgSpec findOriginal(ArgSpec argSpec) {
```

Comment:
Can we make this method package protected instead of public?

Thinking about this some more, it may be better to have a method `ArgSpec::ancestor() : ArgSpec` that returns the original `ArgSpec` for inherited args, and `null` for non-inherited args.  We would also add a getter and setter to `ArgSpec.Builder`. This property would be set in `CommandSpec::addSubcommand` when the inherited subcommand is created:

```java
// CommandSpec addSubcommand(String, CommandLine) // line 6133
if (arg.scopeType() == ScopeType.INHERIT) {
    subSpec.add(arg.isOption()
            ? OptionSpec.builder((OptionSpec) arg).inherited(true).ancestor(arg).build()
            : PositionalParamSpec.builder((PositionalParamSpec) arg).inherited(true).ancestor(arg).build());
}
```

... and similarly to `CommandSpec.addOption` and `CommandSpec.addPositional`.
This can be done separately though.
Thoughts?

Reply:
I like this idea very much, especially with your added suggestion to use the name `root()` which matches the API for the CommandSpec. I am happy to add it here as an additional improvement now.

Reply:
Understood. I’ll hold off on merging until you give the go-ahead. 
Thanks a lot for helping out with this!

Reply:
I've added the code snipped that you suggested with a small change:
```
// CommandSpec addSubcommand(String, CommandLine) // line 6133
if (arg.scopeType() == ScopeType.INHERIT) {
    subSpec.add(arg.isOption()
            ? OptionSpec.builder((OptionSpec) arg).inherited(true).ancestor(arg.root()).build()
            : PositionalParamSpec.builder((PositionalParamSpec) arg).inherited(true).ancestor(arg.root()).build());
}
```
since if we just pass on the current one, we will face the same problem again if we are nesting more than one level deep (which is my case). There is only one positional parameter that is being inherited from the root, but it would still be a problem I think, and thanks to your spot-on definition of the `.root()` call pointing to the `ArgSpec` itself if it is at the root of the tree, it will always work (along with a small check in the initialization of the `ArgSpec` to set the `root` field to itself if it is coming from the `Builder<T>` as `null` and has the scope set to INHERIT. Let me know what you think!

Reply:
I'll mark this as resolved, but I'm open to other comments to other parts of the code if something isn't 100% in line with the project's practices or styles.